### ISSUE TITLE
✨ Convert skill filter in candidate results from intersection to additive

### ISSUE THREAD

**Issue by Jerryescandon** [GCTC-NTGC/gc-digital-talent#6367]
## ✨ Feature

When looking at the results of candidates inside individual request it would be helpful to know which candidates have which of the skills selected in the skill filter.

## 🕵️ Details

Change the way skill filters work instead of intersection (return only users who match ALL skills) return the summation instead. (return ANY user who matches ANY of the skills selected)

In addition to this, adding a column to the candidate results that shows users from MOST skill matches to LEAST skill matches would help DCM decide which candidates fit best to the request.

## 🎨 Design File

[Link to figma](https://www.figma.com/file/UsrB15yicgDCLDQmx3snyl/User-Flows?node-id=3045%3A4807&t=zbLzQ0dXanipnzTx-1)

## 📸 Screenshot

![image](https://user-images.githubusercontent.com/16977254/233731687-ee529022-8d51-417f-be1c-012fa255bf4e.png)

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Criteria 1
- [ ] Criteria 2

## 🛑 Blockers

Blocked by {issue numbers}

**Comment by tristan-orourke** [comment]
I split off a new issue for ordering candidates by skill count #6395. This ticket is now a simpler one, to just change how the filter works.
